buildlib/pr/cuda_compatible.yml

```diff
@@ -0,0 +1,114 @@
+jobs:
+  - job: ucx_build
+    displayName: Build with cuda
+    pool:
+      name: MLNX
+      demands:
+        - ucx_gpu_test -equals yes
+    container: centos7_cuda11
+    steps:
+      - checkout: self
+        clean: true
+        fetchDepth: 100
+
+      - bash: |
+          set -eEx
+          ./autogen.sh
+          mkdir pkg-build
+          cd pkg-build
+          ../contrib/configure-release --with-cuda --with-java=no --prefix=$(Build.Repository.LocalPath)/install
+        displayName: Configure
+
+      - bash: |
+          set -eEx
+          cd pkg-build
+          make -s -j 4
```

Comment:
use `nproc` instead of 4

Reply:
`nproc` take all cores from the system

Reply:
it takes according to agent process cpu affinity